pydantic/v1/schema.py

```diff
@@ -333,7 +333,10 @@ def get_model_name_map(unique_models: TypeModelSet) -> Dict[TypeModelOrEnum, str
     name_model_map = {}
     conflicting_names: Set[str] = set()
     for model in unique_models:
-        model_name = normalize_name(model.__name__)
+        try:
+            model_name = model.model_config["model_name"]
+        except AttributeError:
+            model_name = normalize_name(model.__name__)
```

Comment:
I could try something like this if it is preferable:
```suggestion  
        model_name = (
            model.model_config["model_name"] if not isinstance(model, Enum)
            else normalize_name(model.__name__)
        )
```